### DISCUSSION TITLE
Auto install not working

### DISCUSSION THREAD

**Discussion by mlandsfeld** [tethysplatform/tethys#756] [Q&A]
Hello,
I am developing on a Mac and changes I made to my python scripts used to automatically be installed after I saved the file. This is no longer working. Is there something I can do to revive this? If not, is a quick fix to use "tethys install" I have restarted the machine without success. 

Also, is there a way to see which directory is currently being monitored? I have a couple directories and I am not sure if which one I installed from before the holiday break.
Thanks!

**Comment by swainn** [comment]
Hey @mlandsfeld I'm not sure exactly what you mean by "automatically installed". There are two things I can think of:

1. When developing on an app, you want to have it be installed in "develop" mode so that Python can pick up on the changes. To do this use the `-d` option when you run `tethys install`:

```
tethys install -d
```

2. The development server you use to develop apps (`tethys manage start`) should automatically restart whenever a Python file is changed. You will need to have your app installed in develop mode for this to work (see above). Here's a few more details on this:

- The development server does not need to restart for it to pick up on changes to static files (CSS and JavaScript files) or templates (HTML). However, you may need to do a hard page refresh to get the changes in these files to show up in the web browser (CTRL-SHIFT-R or CMD-SHIFT-R for many web browsers).
- Sometimes while you are editing a Python file, you may introduce an error that causes Tethys to not load the app after the development server restarts. In this case, it won't pick up on any changes to that app anymore, because the app isn't loaded, even if you fix the change that broke it. Simply stop the development server (CTRL-C) and the start it again `tethys manage start` to get it watching your app again.

**Comment by mlandsfeld** [comment]
That fixed it. Thanks! And thanks for the extra tips Nathan!
